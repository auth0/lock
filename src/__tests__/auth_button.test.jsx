import React from 'react';
import { mount } from 'enzyme';

import { expectComponent } from 'testUtils'

import AuthButton from 'ui/button/auth_button';

describe('AuthButton', () => {
  const defaultProps = {
    label: 'label',
    onClick: jest.fn(),
    strategy: 'strategy'
  };
  it('renders correctly', () => {
    expectComponent(
      <AuthButton {...defaultProps} />
    ).toMatchSnapshot();
  });
  it('renders with style customizations', () => {
    expectComponent(
      <AuthButton
        {...defaultProps}
        icon="test"
        primaryColor="primaryColor"
        foregroundColor="foregroundColor"
        />
    ).toMatchSnapshot();
  });
  it('renders when `big` is false', () => {
    expectComponent(
      <AuthButton
        {...defaultProps}
        isBig={false}
        />
    ).toMatchSnapshot();
  });
  it('should trigger onClick when clicked', () => {
    const wrapper = mount(<AuthButton {...defaultProps} />);
    wrapper.find('button').simulate('click');
    expect(defaultProps.onClick.mock.calls.length).toBe(1);
  });
});
