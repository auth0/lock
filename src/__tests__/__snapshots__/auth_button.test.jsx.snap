// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton renders correctly 1`] = `
<button
  className="auth0-lock-social-button auth0-lock-social-big-button"
  data-provider="strategy"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  <div
    className="auth0-lock-social-button-icon"
    style={Object {}}
  />
  <div
    className="auth0-lock-social-button-text"
    style={Object {}}
  >
    label
  </div>
</button>
`;

exports[`AuthButton renders when \`big\` is false 1`] = `
<button
  className="auth0-lock-social-button"
  data-provider="strategy"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  <div
    className="auth0-lock-social-button-icon"
    style={Object {}}
  />
  <div
    className="auth0-lock-social-button-text"
    style={Object {}}
  >
    label
  </div>
</button>
`;

exports[`AuthButton renders with style customizations 1`] = `
<button
  className="auth0-lock-social-button auth0-lock-social-big-button"
  data-provider="strategy"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "primaryColor",
    }
  }
  type="button"
>
  <div
    className="auth0-lock-social-button-icon"
    style={
      Object {
        "backgroundImage": "url('test')",
      }
    }
  />
  <div
    className="auth0-lock-social-button-text"
    style={
      Object {
        "color": "foregroundColor",
      }
    }
  >
    label
  </div>
</button>
`;
